pkg/controller/remotemachineset/awsactuator.go

```diff
@@ -204,3 +213,22 @@ func (a *AWSActuator) updateProviderConfig(machineSet *machineapi.MachineSet, in
 		Value: &runtime.RawExtension{Object: providerConfig},
 	}
 }
+
+// getSubnetsByAvailabilityZones maps availability zones to subnet
+func (a *AWSActuator) getSubnetsByAvailabilityZone(subnets []string) (map[string]string, error) {
+	subnetsByAvailabilityZone := map[string]string{}
+	idPointers := make([]*string, len(subnets))
+	for _, id := range subnets {
+		idPointers = append(idPointers, aws.String(id))
+	}
+
+	results, err := a.client.DescribeSubnets(&ec2.DescribeSubnetsInput{SubnetIds: idPointers})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, subnet := range results.Subnets {
+		subnetsByAvailabilityZone[*subnet.AvailabilityZone] = *subnet.SubnetId
```

Comment:
It should be an error if there has already been a subnet found for the availability zone. This, too, is something that would be good to show in a condition.

Reply:
Addressed

Reply:
Added a condition in the machinepool for this error